### PR TITLE
Issues/8062 consolidate keys

### LINKED ISSUE #8062: Login: Reuse one set of google keys in the app credentials file.
In our app credential files, we currently include Google ID and Scheme keys for development/production builds, internal builds, and alpha builds.  Let's tweak this so there is only one ID and Scheme key and just update the values used for the respective build. 

h/t @SergioEstevao 
cc @nheagy

### PR DESCRIPTION
**Pull request by aerych**
Closes #8062 
This PR consolidates the constants for various sets of keys and secrets used for Google Sign-up into a single set.  Individual build targets should update their own `.wpcom[_*]_app_credentials` files accordingly.

I tried this in the simulator with debug and internal build and everything seemed to work fine for me (mostly). 
I *was* getting an odd error compiling the internal build -- `gencredentials.rb` seemed to think it was encountering a nil `value` when parsing its credentials file. My copy of `.wpcom_internal_app_credentials` was identical to my copy of `.wpcom_app_credentials` with the excpetion of the values so I don't know what was going on there.  I added an nil check to the ruby script and that's sorted it.
I am still getting a warning in the alpha build about missing your custom google url scheme when tapping the google sign-in button. I've had zero luck sorting it ... to my eyes the code changes to the plist preprocessor looks correct and the internal build is *not* having this issue for me.  I'm tempted to call it a caching issue, but I also might just be blind to something I've mis-configured. Hoping another set of eyes can either verify it works or else spot the glitch.

To test:
Build and test debug, release, alpha, and internal builds. Confirm that:
- Builds are successful
- There are no crashes related to the affected constants after the change
- The google login feature works as expected

Summoning multiple sets of eyes for this one since I'm tinkering with unfamiliar things and one of the builds is being problematic, seemingly without rhyme or reason.

Needs review: @nheagy @elibud  and @diegoreymendez

### CODE REVIEW

**Comment by diegoreymendez** [comment]
@aerych - Moving this to 8.9 temporarily to move ahead with the 8.8 code freeze.

That said, if you still want to merge this into 8.8 can you retarget it to `release/8.8` and change the milestone back?

Thanks and sorry for the inconvenience!

**Comment by aerych** [comment]
@diegoreymendez no worries, this is a patch that's mostly for our own convenience so the target milestone doesn't really matter.

**Comment by elibud** [comment]
@aerych I won't be able to properly review this for a few days, but I like the way it looks :) We should test Google login on a TestFlight and an Internal build to be confident before merging.

File: WordPress/WordPress.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6134,7 +6134,7 @@
...
+			shellScript = "[ -f ~/.wpcom_app_credentials ] && source ~/.wpcom_app_credentials\n touch Info.plist\n echo \"\" > InfoPlist.h\n echo \"\" > InfoPlist-alpha.h\n echo \"\" > InfoPlist-internal.h\n if [ \"x$CRASHLYTICS_API_KEY\" != \"x\" ]; then\n echo \"#define FABRIC_API_KEY $CRASHLYTICS_API_KEY\" >> InfoPlist.h\n echo \"#define FABRIC_API_KEY $CRASHLYTICS_API_KEY\" >> InfoPlist-alpha.h\n echo \"#define FABRIC_API_KEY $CRASHLYTICS_API_KEY\" >> InfoPlist-internal.h\n else\n echo \"warning: Crashytics API Key not found\"\n fi\n if [ \"x$GOOGLE_LOGIN_SCHEME_ID\" != \"x\" ]; then\n echo \"#define GOOGLE_LOGIN_SCHEME_ID $GOOGLE_LOGIN_SCHEME_ID\" >> InfoPlist.h\n echo \"#define GOOGLE_LOGIN_SCHEME_ID $GOOGLE_LOGIN_SCHEME_ID\" >> InfoPlist-alpha.h\n echo \"#define GOOGLE_LOGIN_SCHEME_ID $GOOGLE_LOGIN_SCHEME_ID\" >> InfoPlist-internal.h\n else\n echo \"warning: Google Client ID not found\"\n fi";
```
Comment (diegoreymendez) [inline WordPress/WordPress.xcodeproj/project.pbxproj]: Unfortunately I think we will need to move this script into the "WordPress" target.

The reason is that the first line of this script has the credential file hardcoded into it, which isn't good.  The configuration for this file should be taken from the "WPCOM_CONFIG" build configuration setting in target "WordPress" (which is simply inaccessible for other targets).

If you want I can make this change, test it, and add it to the PR.

**Comment by elibud** [comment]
Moving to milestone 9.0 as we'll want Diego to review the latest changes before merging.

**Comment by aerych** [comment]
Thanks @diegoreymendez. The issue with the preprocessor macro was well spotted.  I've opted to go the route @nheagy suggested and promote `WPCOM_CONFIG` to the project scope. I've updated the plist preprocesor macro accordingly.  I'm not too familiar with the scripting syntax but I *think* its correct.  I've retested with develop and internal builds.  No errors as far as I've seen and I can successfully authenticate via Google Sign in in both builds.

Ready for another peek.

**Comment by diegoreymendez** [comment]
@aerych - I failed at going the route I proposed, so I'm glad a new version is up.

I'll review first thing tomorrow.

**Comment by diegoreymendez** [comment]
As discussed previously, I'm taking care of merging this.

**Pull request merged**